custom/icds_reports/utils/aggregation_helpers/ccs_record_monthly.py

```diff
@@ -52,62 +52,139 @@ def tablename(self):
+        age_in_months_end = "({} / 30.4 )".format(age_in_days)
+        open_in_month = ("({} - case_list.opened_on::date)::integer >= 0"
+                         " AND (case_list.closed = 0"
+                         " OR (case_list.closed_on::date - {})::integer > 0)").format(end_month_string,
+                                                                                       start_month_string)
+        alive_in_month = "(case_list.date_death is null OR case_list.date_death-{}>0)".format(
+            start_month_string)
+        seeking_services = "(case_list.is_availing = 1 AND case_list.is_migrated = 0)"
+        ccs_lactating = "({} AND {} AND case_list.add is not null AND {}-case_list.add>=0" \
+                        " AND {}-case_list.add<=183)".format(open_in_month, alive_in_month,
+                                                             end_month_string, start_month_string)
+        lactating = "({} AND {})".format(ccs_lactating, seeking_services)
+        lactating_all = "({} AND  case_list.is_migrated=0)".format(ccs_lactating)
+
+        ccs_pregnant = "({} AND {} AND case_list.edd is not null and" \
+                       " (case_list.add is null OR case_list.add>{}))".format(open_in_month, alive_in_month, end_month_string)
+
+        pregnant_to_consider = "({} AND {} AND {} AND {})".format(ccs_pregnant, seeking_services,
+                                                                  open_in_month, alive_in_month)
+
+        pregnant_all = "({} AND  case_list.is_migrated=0)".format(ccs_pregnant)
+
+        valid_in_month = "( {} OR {})".format(pregnant_to_consider, lactating)
+
+        add_in_month = "(case_list.add>= {} AND case_list.add<={})".format(start_month_string, end_month_string)
+        delivered_in_month = "({} AND {})".format(seeking_services, add_in_month)
+        extra_meal = "(agg_bp.eating_extra=1 AND {})".format(pregnant_to_consider)
+        b1_complete = "(case_list.bp1_date <= {})".format(end_month_string)
+        b2_complete = "(case_list.bp2_date <= {})".format(end_month_string)
+        b3_complete = "(case_list.bp3_date <= {})".format(end_month_string)
+
+        trimester = "(CASE WHEN {} THEN  CASE WHEN (case_list.edd-{})::integer>=183" \
+                    "THEN 1 ELSE CASE WHEN (case_list.edd-{})::integer<91" \
+                    "THEN 3 ELSE 2 END END ELSE null END)".format(pregnant_to_consider, end_month_string,
+                                                                  end_month_string)
+
+        registration_trimester = "(CASE WHEN (case_list.edd-case_list.opened_on::date)::integer>=183" \
+                                 "THEN 1 ELSE CASE WHEN (case_list.edd-case_list.opened_on::date)::integer<91" \
+                                 "THEN 3 ELSE 2 END END)"
+
+        postnatal = "(case_list.add is not null AND ({}-case_list.add)::integer>=0 AND " \
+                    "({}-case_list.add)::integer<=21)".format(end_month_string, start_month_string)
```

Comment:
what is this indicator?

Reply:
hey this is mother till 21 days after delivery
https://docs.google.com/spreadsheets/d/1RDLNIHF3iRkiQtWyrQjKH6f1bStF_S_9woXeITiMJKg/edit#gid=551271206&range=B62